contracts/liquidity_hub/fee_distributor/src/commands.rs

```diff
@@ -31,7 +31,14 @@ pub fn create_new_epoch(deps: DepsMut, env: Env) -> Result<Response, ContractErr
     let start_time =
         if current_epoch.id == Uint64::zero() && current_epoch.start_time == Timestamp::default() {
             // if it's the very first epoch, set the start time to the genesis epoch
-            Timestamp::from_nanos(config.epoch_config.genesis_epoch.u64())
+            let genesis_epoch_timestamp =
+                Timestamp::from_nanos(config.epoch_config.genesis_epoch.u64());
+
+            if env.block.time.nanos() < genesis_epoch_timestamp.nanos() {
```

Comment:
Looks like you are saving 
`genesis_epoch_timestamp ` as a Timestamp too so you might not need Nanos when you instantiate but you can still convert to nanos from here

Reply:
This could indeed be `env.block.time.seconds() < genesis_epoch_timestamp.seconds()`, but for sake of consistency I just use nanos there.